openssl/src/x509/mod.rs

```diff
@@ -1472,6 +1511,46 @@ impl X509RevokedRef {
             Asn1IntegerRef::from_ptr(r as *mut _)
         }
     }
+
+    /// Get the issuer name of the revoked certificate
+    #[corresponds(X509_REVOKED_get_ext_d2i)]
+    pub fn issuer_name(&self) -> Option<Stack<GeneralName>> {
+        // SAFETY: self.as_ptr() is a valid pointer to an X509_REVOKED.
+        unsafe {
+            let issuer_names = ffi::X509_REVOKED_get_ext_d2i(
+                self.as_ptr() as *const _,
+                // NID_certificate_issuer is a X509_REVOKED extension that
+                // returns a GENERAL_NAMES, which is a Stack<GeneralName>
+                ffi::NID_certificate_issuer,
+                // Only one instance of the extension is permissable
+                ptr::null_mut(),
+                // Don't care if the extension is critical
+                ptr::null_mut(),
+            );
+            Stack::from_ptr_opt(issuer_names as *mut _)
+        }
+    }
+
+    /// Get the reason that the certificate was revoked
+    #[corresponds(X509_REVOKED_get_ext_d2i)]
+    #[cfg(ossl110)]
+    pub fn reason_code(&self) -> Option<Result<CrlReason, ErrorStack>> {
+        let reason_code = unsafe {
+            // The return value may be NULL if the extension wasn't found or
+            // there were multiple, and we require only one.
+            Asn1Enumerated::from_ptr_opt(ffi::X509_REVOKED_get_ext_d2i(
+                // self.as_ptr() is a valid pointer to a X509_REVOKED
+                self.as_ptr() as *const _,
+                // NID_crl_reason is an X509_REVOKED extension that is an ASN1_ENUMERATED
+                ffi::NID_crl_reason,
+                // Only one instance of the extension is permissable
+                ptr::null_mut(),
+                // Don't care if the extension is critical
+                ptr::null_mut(),
+            ) as *mut _)
+        }?;
+        Some(reason_code.get_i64().map(CrlReason::from_raw))
```

Comment:
I'm not sure how I feel about this doing the conversion to CrlReason internally.

Reply:
I like it as it feels a little more type-safe with no downsides I can see (CRLReason has a from_raw and to_raw). It's moot now though as the trait based system can't do it.